mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -508,12 +508,10 @@ class LogisticRegressionModel private[spark] (
    * thrown if `trainingSummary == None`.
    */
   @Since("1.5.0")
-  def summary: LogisticRegressionTrainingSummary = trainingSummary match {
-    case Some(summ) => summ
-    case None =>
-      throw new SparkException(
-        "No training summary available for this LogisticRegressionModel",
-        new NullPointerException())
+  def summary: LogisticRegressionTrainingSummary = trainingSummary.getOrElse {
+    throw new SparkException(
+      "No training summary available for this LogisticRegressionModel",
+      new RuntimeException())
```

Comment:
Yes, I agree we don't use ```RuntimeException``` directly. But I think it's inappropriate use ```NullPointerException``` here, because ```NullPointerException``` was used when an application attempts to access null in a case. Here we throw the ```SparkException``` just want to handle the case that ```trainingSummary == None```. Should we just throw an ```SparkException```?